src/systems/ec/implementations/react/packages/checkbox/Checkbox.jsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import Icon from '@ecl/ec-react-component-icon/Icon';
+
+const Checkbox = ({
+  className,
+  defaultChecked,
+  disabled,
+  id,
+  helperText,
+  hideLabel,
+  invalid,
+  invalidText,
+  label,
+  name,
+  ...props
+}) => {
+  const classNames = classnames(className, 'ecl-checkbox', {
+    'ecl-checkbox--invalid': invalid,
+  });
+
+  return (
+    <div className="ecl-form-group ecl-form-group--checkbox">
+      <div className={classNames}>
+        <input
+          {...props}
+          type="checkbox"
+          id={id}
+          defaultChecked={defaultChecked}
+          name={name || undefined}
+          className="ecl-checkbox__input"
+          disabled={disabled}
+        />
+        <span className="ecl-checkbox__icon">
+          <Icon className="ecl-checkbox__checked" shape="ui--check" size="xs" />
+        </span>
+
+        <label
+          className={classnames(
+            'ecl-form-label ecl-form-label--inline ecl-checkbox__label',
+            {
+              'ecl-form-label--invalid': invalid,
+              'ecl-form-label--hidden': hideLabel,
```

Comment:
hidding the label breaks the display of the box:
![image](https://user-images.githubusercontent.com/25579221/51251075-b0983800-1998-11e9-8130-1a6d2321d535.png)

Reply:
Yeah, now that we have the icon inside the label the problem shifted, but it was still there, meaning that with an hidden label we would have had also the fake checkbox hidden. Fixed by overriding the behaviour of the hidden label in the checkbox component.